CMakeLists.txt

```diff
@@ -42,9 +43,24 @@ set(C_WARNING_Clang -Wall -Wextra -Werror -Wpedantic
         -Wstrict-prototypes -Wold-style-definition
         -Wno-unused-parameter -Wno-missing-field-initializers -Wno-sign-compare -Wno-language-extension-token)
 
-# Additional compiler-specific flags
-#  -gdwarf-aranges to workaround https://sourceware.org/bugzilla/show_bug.cgi?id=22288 (backtrace line printing with libdw and clang)
 set(C_EXTRA_GNU )
+if (ENABLE_LINKTIME_OPTIMIZATION)
+    include(CheckIPOSupported)
+    check_ipo_supported(RESULT ipo_supported OUTPUT output)
+    if(ipo_supported)
+      message(STATUS "Interprocedural optimization is supported")
+      set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
+      set(C_EXTRA_GNU -fno-fat-lto-objects)
```

Comment:
Isn't this the default? Judging from https://gitlab.kitware.com/cmake/cmake/-/issues/21696 and https://gitlab.kitware.com/cmake/cmake/-/blob/eaea031fb01d0c97b132e9809702da7eb0880567/Modules/Compiler/GNU.cmake#L99

I guess I would not worry about the flag either way and just let CMake to set its default LTO flags. This seems possibly important for libraries, and the router is not one.

Reply:
Done ! Thanks.